stellargraph/layer/knowledge_graph.py

```diff
@@ -191,3 +191,126 @@ def build(self):
         x_out = self(x_inp)
 
         return x_inp, x_out
+
+
+class DistMultScore(Layer):
+    """
+    DistMult scoring Keras layer.
+
+    Original Paper: Embedding Entities and Relations for Learning and Inference in Knowledge
+    Bases. Bishan Yang, Wen-tau Yih, Xiaodong He, Jianfeng Gao, Li Deng. ICLR 2015
+
+    This combines subject, relation and object embeddings into a score of the likelihood of the
+    link.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def build(self, input_shape):
+        self.built = True
+
+    def call(self, inputs):
+        e1, r, e2 = inputs
+        # y_(e_1)^T M_r y_(e_2), where M_r = diag(w_r) is a diagonal matrix
+        score = tf.reduce_sum(e1 * r * e2, axis=2)
+        return score
+
+
+@experimental(reason="results from the reference paper have not been reproduced yet")
+class DistMult:
+    """
+    Embedding layers and a DistMult scoring layers that implement the DistMult knowledge graph
+    embedding algorithm as in https://arxiv.org/pdf/1412.6575.pdf
+
+    Args:
+        generator (KGTripleGenerator): A generator of triples to feed into the model.
+
+        k (int): the dimension of the embedding (that is, a vector in R^k is learnt for each node
+            and each link type)
+
+        embedding_initializer (str or func, optional): The initialiser to use for the embeddings.
+
+        embedding_regularizer (str or func, optional): The regularizer to use for the embeddings.
+    """
+
+    def __init__(
+        self, generator, k, embedding_initializer=None, embedding_regularizer=None,
```

Comment:
to keep things consistent with `ComplEx` and `tensorflow` I think there should 's' on `embedding`:

```suggestion
        self, generator, k, embeddings_initializer=None, embeddings_regularizer=None,
```

Reply:
Oh, I forgot to update this after doing it for `ComplEx`. Thanks.